src/main/java/seedu/address/logic/commands/CompleteCommand.java

```diff
@@ -0,0 +1,139 @@
+            + "by the index number used in the displayed person list or "
+            + "by a specified date"
+            + "Parameters: INDEX(must be a positive integer) "
+            + "[" + PREFIX_APPOINTMENT_DATE + "Appointment Date] "
+            + "Example: " + COMMAND_WORD + " 1 ";
+    public static final String MESSAGE_COMPLETE_SUCCESS = "Appointments Completed!";
+    public static final String MESSAGE_NOT_COMPLETED = "At least one identifier must be provided.";
+    public static final String MESSAGE_INVALID_DATE_FORMAT = "Input Date should be in format of dd-MM-yyyy";
+    public static final String MESSAGE_INVALID_DATE = "Please input a valid Date";
+    private final CompleteDescriptor completeDescriptor;
+    public CompleteCommand(CompleteDescriptor completeDescriptor) {
+        this.completeDescriptor = completeDescriptor;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Consumer<Index> editByIndex = index -> {
+            Person personToEdit = lastShownList.get(index.getZeroBased());
+            Person edittedPerson = createPersonWithoutAppointment(personToEdit);
+
+            model.setPerson(personToEdit, edittedPerson);
+        };
+
+        Consumer<LocalDate> editByDate = date -> {
+            model.clearAppointments(date);
+        };
+
+        Optional<Index> index = completeDescriptor.getIndex();
+        if (index.isPresent()) {
+            if (index.get().getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+        }
+
+        // if Index present, edit Model by Person Index
+        completeDescriptor.getIndex().ifPresent(editByIndex);
+
+        // if Date present, edit Model by Date;
+        completeDescriptor.getDate().ifPresent(editByDate);
+
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(MESSAGE_COMPLETE_SUCCESS);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof CompleteCommand)) {
+            return false;
+        }
+
+        CompleteCommand otherEditCommand = (CompleteCommand) other;
+        return completeDescriptor.equals(otherEditCommand.completeDescriptor);
+    }
+
+    private static Person createPersonWithoutAppointment(Person personToEdit) {
+        return new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getAddress(), personToEdit.getNextOfKinName(), personToEdit.getNextOfKinPhone(),
+                personToEdit.getFinancialPlans(), personToEdit.getTags(), NullAppointment.getNullAppointment());
+    }
+
```

Comment:
Is there a reason this is here? seems like its not necessary unless testing?

Reply:
Person fields are all final, the only way to change Appointment to a NullAppointment is to create a copy of the Person and use the model setPerson method